specification/redhatopenshift/resource-manager/Microsoft.RedHatOpenShift/stable/2020-04-30/redhatopenshift.json

```diff
@@ -0,0 +1,778 @@
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/OpenShiftClusterCredentials"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.  If the resource doesn't exist, 404 (Not Found) is returned.  If any of the input parameters is wrong, 400 (Bad Request) is returned.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Lists credentials of an OpenShift cluster with the specified subscription, resource group and resource name.": {
+            "$ref": "./examples/OpenShiftClusters_ListCredentials.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "APIServerProfile": {
+      "description": "APIServerProfile represents an API server profile.",
+      "properties": {
+        "visibility": {
+          "$ref": "#/definitions/Visibility",
+          "description": "API server visibility (immutable)."
+        },
+        "url": {
+          "description": "The URL to access the cluster API server (immutable).",
+          "type": "string"
+        },
+        "ip": {
+          "description": "The IP of the cluster API server (immutable).",
+          "type": "string"
+        }
+      }
+    },
+    "CloudError": {
+      "description": "CloudError represents a cloud error.",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/CloudErrorBody",
+          "description": "An error response from the service."
+        }
+      }
+    },
+    "CloudErrorBody": {
+      "description": "CloudErrorBody represents the body of a cloud error.",
+      "properties": {
+        "code": {
+          "description": "An identifier for the error. Codes are invariant and are intended to be consumed programmatically.",
+          "type": "string"
+        },
+        "message": {
+          "description": "A message describing the error, intended to be suitable for display in a user interface.",
+          "type": "string"
+        },
+        "target": {
+          "description": "The target of the particular error. For example, the name of the property in error.",
+          "type": "string"
+        },
+        "details": {
+          "description": "A list of additional details about the error.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CloudErrorBody"
+          }
+        }
+      }
+    },
+    "ClusterProfile": {
+      "description": "ClusterProfile represents a cluster profile.",
+      "properties": {
+        "pullSecret": {
+          "description": "The pull secret for the cluster (immutable).",
+          "type": "string"
+        },
```

Comment:
secrets should only be exposed using POST actions.  Does the user provide this?  If it is provided it can be here but should not be returned after it is set (like adminPassword in VMs)

Reply:
Yes, the user provides it, and it is not returned after being set.  The examples also reflect this.